packages/babel-plugin-transform-es2015-block-scoping/src/index.js

```diff
@@ -416,56 +416,79 @@ class BlockScoping {
     // build the closure that we're going to wrap the block with, possible wrapping switch(){}
     const fn = t.functionExpression(null, params,
       t.blockStatement(isSwitch ? [block] : block.body));
-    fn.shadow = true;
```

Comment:
This whole set of changes is gross and probably the most likely to be missing edge cases, but it works as far as I can tell.

These changes were all needed so that we can get access to the final NodePath for the function, since it wasn't available anywhere before.